scripts/deploy/deployment_file.py

```diff
@@ -124,27 +124,39 @@ def update_contract_deployment(
         else:
             encoded_args = None
 
+        with open(BASE_DIR / contract_path) as f:
+            source_code = f.read()
+
         # fetch data from contract pragma:
         pattern = r"# pragma version ([\d.]+)"
-        match = re.search(pattern, contract_object.compiler_data.source_code)
+        match = re.search(pattern, source_code)
         if match:
             compiler_version = match.group(1)
         else:
             raise ValueError("Compiler Version is set incorrectly")
 
         # latest git commit hash:
         latest_git_commit_for_file = get_latest_commit_hash(contract_object.filename)
-        contract_relative_path = get_relative_path(contract_object.filename)
+        contract_relative_path = get_relative_path(contract_path)
         github_url = (
             f"https://github.com/curvefi/curve-lite/blob/{latest_git_commit_for_file}/"
             f"{'/'.join(contract_relative_path.parts[1:])}"
         )
 
+        optimisation_level = "GAS"  # it's not in new version of boa, backward compatibility
+        evm_version = "shanghai"
+
+        # fetch data from contract pragma:
```

Comment:
here i think you should instead enforce there to be something in the pragma instead of doing regex pattern matching.